train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -41,15 +41,30 @@ private void enforceSpeedLimit() {
 		}
 	}
 
+	@Override
+    public void run() {
+        while (true) {
+            try {
+                // Call the followSpeed method
+                this.followSpeed();
+                Thread.sleep(10); // Sleep for 10 milliseconds
```

Comment:
this might lead to too fast acceleration, consider changing it to 100 or 1000 ms